Document contract update validation criteria (#793)
onflow/cadence: docs/language/contracts.md

> 🚧 Status: Updating contracts is **experimental**.
>

> > Updating contracts currently have limited support to maintain data consistency.
> > Read more details on possible changes and restrictions imposed on updating contracts in the
> > [contract updatability](../contract-updatability.md) section.

Reviewer: ```suggestion
> Updating contracts is currently limited to maintain data consistency.
> Read more details on valid changes and restrictions imposed on updating contracts in the
> [contract updatability](../contract-updatability) section.
```
